rust/src/lib/unit_tests/debug_trait.rs

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Interface, NetworkState};
+
+#[test]
+fn test_debug_trait() {
+    // Ref: <PROJECT_ROOT>/rust/src/lib/ieee8021x.rs Line: 8
+    let ethernet = serde_yaml::from_str(
+        r"
+        name: eth1
+        type: ethernet
+        state: up
+        802.1x:
+            ca-cert: /etc/pki/802-1x-test/ca.crt
+            client-cert: /etc/pki/802-1x-test/client.example.org.crt
+            eap-methods:
+                - tls
+            identity: client.example.org
+            private-key: /etc/pki/802-1x-test/client.example.org.key
+            private-key-password: A1234567890
+        ",
+    )
+    .unwrap();
+
+    let des_eth = Interface::base_iface(&ethernet);
+
+    let mut net_state = NetworkState::new();
+    net_state.interfaces.push(ethernet.clone());
+
+    // println!("{:?}\n\n", &des_eth);
+    // println!("{:?}\n\n", &net_state);
+
+    let debug_tr_eth = format!("{:?}", &des_eth);
+    let debug_tr_net_state = format!("{:?}", &net_state);
+
+    let password_hid_by_nmstate =
+        format!("private_key_password: Some(\"<_password_hid_by_nmstate>\")");
+
+    let miss_matched_pass =
+        des_eth.ieee8021x.as_ref().unwrap().private_key_password
+            == Some("A123456789000".to_string());
+    let matched_pass = des_eth.ieee8021x.as_ref().unwrap().private_key_password
+        == Some("A1234567890".to_string());
+    let hidden_in_eth = debug_tr_eth.contains(&password_hid_by_nmstate);
+    let hidden_in_net_state =
+        debug_tr_net_state.contains(&password_hid_by_nmstate);
+
+    assert!(!miss_matched_pass);
+    assert!(matched_pass);
+    assert!(hidden_in_eth);
+    assert!(hidden_in_net_state);
```

Comment:
No need to assert on boolean, just use `assert!()` on above functions, it could be better for developer to understand the purpose of checking.

Reply:
I don't think i understand what u mean by "just use assert!() on above functions"